Copied IString code from v14.9.0

ehoogerbeets opened the pull request:
* all locale data is scrunched together in allplurals.json
* converted to ES6
* now supports spread operator and for..of construct through the @@Iterator generator function
* updated to delegate all methods to intrinsic string
* updated to use jsdoc v3

ehoogerbeets committed:
Copied from v14.9.0

gouniLee commented on locale/allplurals.json:
```diff
@@ -0,0 +1 @@
+{"af":{"one":{"eq":["n",1]}},"ak":{"one":{"eq":["n",[0,1]]}},"am":{"one":{"or":[{"eq":["i",0]},{"eq":["n",1]}]}},"an":{"one":{"eq":["n",1]}},"ar":{"zero":{"eq":["n",0]},"one":{"eq":["n",1]},"two":{"eq":["n",2]},"few":{"eq":[{"mod":["n",100]},[3,10]]},"many":{"eq":[{"mod":["n",100]},[11,99]]}},"ars":{"zero":{"eq":["n",0]},"one":{"eq":["n",1]},"two":{"eq":["n",2]},"few":{"eq":[{"mod":["n",100]},[3, ...
```
How are you going to maintain this file? because committed state is  already merged and uglified form

ehoogerbeets replied:
A different PR will come later that will implement a tool based on the ilib cldr tools. This tool will generate this file. Only thing is, the cldr tool 'genplurals.js' is sort of broken right now, so it is going to be a significant amount of work.

gouniLee replied:
What kinds of broken in `genplurals.js`? I just try to run it in the development branch. it generates `plurals.json` without an error  by following the command
`node genplurals.js tmp`

gouniLee commented on package.json:
```diff
@@ -0,0 +1,96 @@
+        "build:prod": "grunt babel --mode=prod",
+        "build:dev": "grunt babel --mode=dev",
+        "build:web": "webpack-cli",
+        "generate": "node generate.js",
```
No generate.js file in this repo

ehoogerbeets replied:
That's the tool that will come later.

gouniLee commented on test/log4js.json:
```diff
@@ -0,0 +1,213 @@
+{
+    "appenders": {
```
I think It's not needed in this repo. no?

ehoogerbeets replied:
yes, I'll delete

gouniLee commented on webpack.config.js:
```diff
@@ -0,0 +1,48 @@
+/*
+ * webpack.config.js - webpack configuration script for ilib-env
```
I think this file is not needed in this repo too.

ehoogerbeets replied:
It will be soon. I'm trying to figure out how to run the es6 tests in the web browser. (Probably need to use babel to convert them to old JS.)

gouniLee reviewed (commented):
This repo should be public. Currently, it's private.

ehoogerbeets committed:
Removed unnecessary file

ehoogerbeets committed:
Optimize the output

- minify
- strip comments

gouniLee commented on test/testSuite.html:
```diff
@@ -27,7 +27,8 @@
     <script src="./nodeunit/nodeunit.js"></script>
```
files in nodeunit/ directory are missed in this PR

ehoogerbeets committed:
Got the web tests working properly

- added testall and build:test scripts in package.json

ehoogerbeets committed:
Some cleanup in package.json

ehoogerbeets committed:
Update dependencies

- make doc output be markdown instead of html
- use Terser instead of UglifyJS webpack plugin, which is deprecated

ehoogerbeets commented:
@gouniLee this is ready for a re-review now
